### ISSUE TITLE
closed tabs list with too many entries

### ISSUE THREAD

**Issue by vbr** [OtterBrowser/otter-browser#992]
Hi,
I'd like to report a minor glitch in the list of last closed tabs - if you manage to fill the list with many entries, which can't fit on the screen any more, it seems, that there remain some items "offscreen", which are not visible, but one can apparently select them with cursor keys and open using Enter.
It's clearly a cornercase, but maybe it could be made more coherent by some kind of scrolling or wrapping the menu in such cases.
(Otter weekly 96, 32bit, Win7).

regards
    vbr

**Comment by Frenzie** [comment]
Sounds like an upstream issue. For me (on Debian Jessie Xfce) a second
column appears once the first is filled up. I'm not sure what happens when
the entire screen is full. I suppose my preferred solution would be some
kind of "more" submenu.

**Comment by vbr** [comment]
Yes, I should have clarified this in my description, the mentioned behaviour appears after the entire screen is filled, i.e. with multiple columns. 
For me too, the "more" submenu item in the last visible position would be expected in such situations, which would open another "layer" of the menu items.
   vbr

**Comment by Emdek** [comment]
Default behavior of ``QMenu`` appears to be causing various issues, so what do you think about changing it to custom solution that would use scrolling buttons on top and bottom instead (plus optional filter field, as noted in another issue)?
``QMenu`` is one of least flexible widgets in Qt, but since it is one of most common used widgets it will be also very hard to replace it (especially if going for widget created from scratch, it will require custom buttons and menu bar), so we would need to do some planning before deciding to go for such radical change.

**Comment by vbr** [comment]
Hi, the proposed solution looks nice, but actually, the problem seems to be a marginal one, as it requires quite special circumstances to appear. You have to balance the cost of this approach both in terms of coding effort and maintainability etc.
regards
   vbr

**Comment by Emdek** [comment]
@vbr, well, other long menu have exactly the same issue, so that would improve situation for more cases. Also there would be other benefits, like ability to add custom items (like Chrome style zoom controls) to menus (as option, depending on menu definition file, as discussed in some older ticket).

**Comment by VSHY** [comment]
In build 98 the list of last closed tabs is not working.

**Comment by Emdek** [comment]
@VSHY, which build?
Is that reproducible?
Note that private tabs won't end up there.
Also please check if they are shown in History menu in such case.

**Comment by VSHY** [comment]
Very strange, I can't reproduce that... Sorry.

**Comment by ZarkBit** [comment]
Regarding the too many entries
https://dl.dropboxusercontent.com/s/e4mkx5gy65ps0fp/2015-11-19_20-19-23.png

2 monitors in use, and there are more closed tabs to the right which cannot be seen.
